src/main/java/org/jabref/logic/openoffice/oocsltext/CSLCitationOOAdapter.java

```diff
@@ -12,55 +14,199 @@
+    private void insertMultipleReferenceMarks(XTextDocument doc, XTextCursor cursor, List<BibEntry> entries, OOText ooText) throws Exception {
+        // Check if there's already a space before the cursor
+        boolean spaceExists = false;
+        XTextCursor checkCursor = cursor.getText().createTextCursorByRange(cursor.getStart());
+
+        // Check if we're at the start of the document - if yes we set the flag and don't insert a space
+        if (!checkCursor.goLeft((short) 1, true)) {
+            // We're at the start of the document
+            spaceExists = true;
+        } else {
+            // If not at the start of document, check if there's a space before
+            spaceExists = checkCursor.getString().equals(" ");
+            // If not a space, check if it's a paragraph break
+            if (!spaceExists) {
+                spaceExists = checkCursor.getString().matches("\\R");
+            }
+        }
 
-        String formattedCitation = transformHtml(citation);
+        if (entries.size() == 1) {
+            CSLReferenceMark mark = markManager.createReferenceMark(entries.getFirst());
+            mark.insertReferenceIntoOO(doc, cursor, ooText, !spaceExists, true, true);
+        } else {
+            if (!spaceExists) {
+                cursor.getText().insertString(cursor, " ", false);
+            }
+            OOTextIntoOO.write(doc, cursor, ooText);
+            for (BibEntry entry : entries) {
+                CSLReferenceMark mark = markManager.createReferenceMark(entry);
+                OOText emptyOOText = OOFormat.setLocaleNone(OOText.fromString(""));
+                mark.insertReferenceIntoOO(doc, cursor, emptyOOText, false, false, true);
+            }
+            cursor.getText().insertString(cursor, " ", false);
+        }
+
+        // Move the cursor to the end of the inserted text
+        cursor.collapseToEnd();
+    }
+
+    private String updateMultipleCitations(String citation, List<BibEntry> entries) {
+        Pattern pattern = Pattern.compile("(\\D*)(\\d+)(\\D*)");
+        Matcher matcher = pattern.matcher(citation);
+        StringBuilder sb = new StringBuilder();
+        int entryIndex = 0;
+
+        while (matcher.find() && entryIndex < entries.size()) {
+            String prefix = matcher.group(1);
+            String suffix = matcher.group(3);
+
+            int currentNumber = markManager.getCitationNumber(entries.get(entryIndex).getCitationKey().orElse(""));
+
+            matcher.appendReplacement(sb, Matcher.quoteReplacement(prefix + currentNumber + suffix));
+            entryIndex++;
+        }
+        matcher.appendTail(sb);
+        return sb.toString();
+    }
+
+    private void writeCitation(XTextDocument doc, XTextCursor cursor, BibEntry entry, String citation) throws Exception {
+        String citationKey = entry.getCitationKey().orElse("");
+        int currentNumber = markManager.getCitationNumber(citationKey);
+
```

Comment:
change method names of this class to more relevant ones (check their flow and utility, maybe simplify so that each method deals with one thing so that we can easily come up with non confusing names)

Reply:
refactoring tasks moved to PR-D